diff --git a/Cargo.toml b/Cargo.toml
index a48b97b60b..c8cb6abdf4 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -322,6 +322,10 @@ buck2_critical_path = { path = "app/buck2_critical_path" }
 [profile.release]
 incremental = true
 panic = "abort"
+lto = "thin"
+codegen-units = 1
+opt-level = 3
+strip = "debuginfo"
 # The line below increases build times from 50s to 84s, with no observed impact on runtime speed.
 # But maybe for some benchmarks it will show more noticeable variation.
 # lto = "thin"
