diff --git a/remote_execution/oss/re_grpc/src/client.rs b/remote_execution/oss/re_grpc/src/client.rs
index a34b006d71..ab22b9d756 100644
--- a/remote_execution/oss/re_grpc/src/client.rs
+++ b/remote_execution/oss/re_grpc/src/client.rs
@@ -157,7 +157,6 @@ impl REClientBuilder {
 
             anyhow::Ok(
                 Channel::from_shared(address.clone())?
-                    .tls_config(tls_config.clone())?
                     .connect()
                     .await
                     .with_context(|| format!("Error connecting to `{}`", address))?,
